src/data/nat/enat.lean

```diff
@@ -361,6 +361,14 @@ begin
   apply_mod_cast nat.lt_succ_of_le, apply_mod_cast h
 end
 
+lemma ne_zero_iff_one_le {n : enat} : n ≠ 0 ↔ 1 ≤ n :=
+begin
+  change ¬ n = ⊥ ↔ _,
+  rw [eq_bot_iff, ← lt_iff_not_le],
+  convert (add_one_le_iff_lt (show (0 : enat) ≠ ⊤, from coe_ne_top 0)).symm,
+  exact (zero_add 1).symm
+end
+
```

Comment:
Watch out -- this file was renamed to `part_enat.lean`. https://mathlib-changelog.org/commit/201d2c6 (@urkud Should we wait to PR this once you've created `enat`? or is `part_enat` fine for now?)

I'm not sure what the right design is here, but with enough simp lemmas this proof becomes a one-liner:
```lean
namespace part_enat

@[simp] lemma coe_eq_zero_iff {x : ℕ} : (x : part_enat) = 0 ↔ x = 0 := coe_inj
@[simp] lemma top_ne_coe (x : ℕ) : ⊤ ≠ (x : part_enat) := (coe_ne_top _).symm
@[simp] lemma zero_ne_top : (0 : part_enat) ≠ ⊤ := coe_ne_top _
@[simp] lemma top_ne_zero : ⊤ ≠ (0 : part_enat) := top_ne_coe _
@[simp, norm_cast] lemma one_le_coe {x : ℕ} : 1 ≤ (x : part_enat) ↔ 1 ≤ x := coe_le_coe

lemma ne_zero_iff_one_le {n : part_enat} : n ≠ 0 ↔ 1 ≤ n :=
part_enat.cases_on n (by simp) (by simp [nat.one_le_iff_ne_zero])

end part_enat
```
(The problem I'm seeing is that these simp lemmas seem to require duplication, one for each numeral you come across, since they're not just coerced nats.)